### PR TITLE
rocPyDecode documentation update

### PR DESCRIPTION
**Pull request by rrawther**
Update to correct API spec

### CODE REVIEW

**Comment by LakshmiKumar23** [comment]
Please add H264 to this: https://github.com/ROCm/rocPyDecode/blob/develop/docs/install/install.rst#supported-codecs
We should also remove Ub20.04 from the same place. The way we install dlpack and pybind11 works only with ub22 onwards

File: docs/how-to/using-rocPydecode.rst [hunk 1]
```diff
@@ -6,155 +6,64 @@
...
 
-The samples use the ``RocVideoDecoder`` user class provided in ``roc_video_dec.h`` in the ``utils`` folder
-of the rocDecode repository.
+For detailed explanation of rocDecode API, please refer to `rocDecode documentation <https://rocm.docs.amd.com/projects/rocDecode/en/latest/>`_.
+The samples uses the ``pyRocVideoDecode`` python module to interface with the low level ``rocVideoDecode`` class available in the C++ rocDecode library.
 
-A video parser (defined in ``rocparser.h``) is needed to extract and decode headers from the bitstream
-in order to organize the data into a structured format for the hardware decoder. The parser is critical in
-video decoding, as it controls the decoding and display of a bitstream's individual frames and fields.
+The ``pyRocVideoDecode`` module exposes the following APIs thorugh two python classes ``PyRocVideoDecoder`` and ``PyVideoDemuxer``.
```
Comment (LakshmiKumar23) [inline docs/how-to/using-rocPydecode.rst]: spell check - "through"

File: docs/how-to/using-rocPydecode.rst [hunk 2]
```diff
@@ -6,155 +6,64 @@
...
 
-The parser object in ``rocparser.h`` has three main APIs:
+class PyRocVideoDecoder (Video decoder class)
+    * ``GetDeviceinfo(self)`` -> API to get device information of the current device
+    * ``SetReconfigParams(self)`` -> API to reconfigure decoder when thre is codec/resolution changes
+    * ``DecodeFrame(self)`` -> API to trigger decoding of a frame thorugh the low level decoder
```
Comment (LakshmiKumar23) [inline docs/how-to/using-rocPydecode.rst]: same as above "through"

File: docs/how-to/using-rocPydecode.rst [hunk 3]
```diff
@@ -6,155 +6,64 @@
...
+    * ``GetFrame(self)`` -> API to receive a decoded output frame in video memory
+    * ``SaveFrameToFile(self)`` -> API to bring back the decoded output froame to host and save it to a file
```
Comment (LakshmiKumar23) [inline docs/how-to/using-rocPydecode.rst]: spell check - "frame"

File: docs/how-to/using-rocPydecode.rst [hunk 4]
```diff
@@ -6,155 +6,64 @@
...
 
-Use the ``rocDecDecodeFrame()`` API to submit a new frame for hardware decoding. Underneath the
+Use the ``DecodeFrame()`` API to submit a new frame for hardware decoding. Underneath the
 driver, the Video Acceleration API (VA-API) is used to submit compressed picture data to the driver.
 The parser extracts all the necessary information from the bitstream and fills the ``RocdecPicParams``
 structure that's appropriate for the codec. The high-level ``RocVideoDecoder`` class connects the parser
 and decoder used for all sample applications.
 
-The ``rocDecDecodeFrame()`` call takes the decoder handle and the pointer to the ``RocdecPicParams``
+The ``DecodeFrame()`` call will call the C++ ``rocDecDecodeFrame()`` call takes the decoder handle and the pointer to the ``RocdecPicParams``
```
Comment (LakshmiKumar23) [inline docs/how-to/using-rocPydecode.rst]: not very clear this statement. Suggestion - "The ``DecodeFrame()`` function will call the C++ ``rocDecDecodeFrame()`` function which takes the decoder handle and the pointer to the ``RocdecPicParams`` as arguments"

File: docs/how-to/using-rocPydecode.rst [hunk 5]
```diff
@@ -6,155 +6,64 @@
...
+successful completion of ``GetFrame()`` indicates that the decoding process is complete and
 the device memory pointer is inter-opped into the ROCm HIP address space in order to further process
-the decoded frame in device memory. The caller gets the necessary information on the output surface,
-such as YUV format, dimensions, and pitch from this call. In the high-level ``RocVideoDecoder`` class, we
-provide four different surface type modes for the mapped surface, as specified in
-``OutputSurfaceMemoryType``.
+the decoded frame in device memory. The caller gets the necessary information of the output surface,
+such as YUV format, dimensions, and pitch by calling the ``GetOutputSurfaceInfo()`` api call. 
+In the high-level ``RocVideoDecoder`` class, we provide four different surface_type modes for the mapped surface, as specified in
+``OutputSurfaceMemoryType``. Please refer to ``rocDecode`` documentation for further details on these modes.
```
Comment (LakshmiKumar23) [inline docs/how-to/using-rocPydecode.rst]: remove quotes for rocDecode. It's not used anywhere else. Looks weird in the documentation

**Comment by rrawther** [comment]
@LakshmiKumar23 : addressed the comments. Please check again

**Review by LakshmiKumar23 — APPROVED** [review]

**Review by kiritigowda — APPROVED** [review]

**Pull request merged**